Import PUG guide to MANIFEST.in (#4108)
pypa/setuptools: docs/userguide/miscellaneous.rst

=====================================

For the most common use cases, ``setuptools`` will automatically find out which
files are necessary for distributing the package. More precisely, the following
files are included in a source distribution by default:

- All Python source files implied by the ``py_modules`` and ``packages``
  ``setup()`` arguments
- All C source files mentioned in the ``ext_modules`` or ``libraries``

>   ``setup()`` arguments

Reviewer: ```suggestion
  ``setup()`` arguments
- Files that match the following glob patterns:
  ``['tests/test*.py', 'test/test*.py']``
```

From https://github.com/pypa/distutils/blob/main/distutils/command/sdist.py#L233-L246

- Scripts specified by the ``scripts`` ``setup()`` argument

> - All files specified by the ``package_data`` and ``data_files``
>   ``setup()`` arguments

Reviewer: ```suggestion
- All files specified by the ``package-data`` and ``data-files``
  configuration parameters in ``pyproject.toml`` and/or equivalent in 
  ``setup.cfg``/``setup.py``.
```

- The file specified by the ``license_file`` option in ``setup.cfg``

>   (setuptools 40.8.0+)

Reviewer: I think it is not worthy to keep track of all the versions directly here.
There will be cases that a parameter is supported since one version in `setup.py`, another version in `setup.cfg` and yet another version in `pyproject.toml`...

Maybe we can add a note at the end saying something like:

> Please note that the list above is guaranteed to work in the last stable version
> of ``setuptools``. The behaviour of older versions might differ.

> - All files specified by the ``license_files`` option in ``setup.cfg``
>   (setuptools 42.0.0+)

Reviewer: ```suggestion
- All files specified by the ``license-files`` configuration parameter
  in ``pyproject.toml`` and/or equivalent in  ``setup.cfg``/``setup.py``.
  Note that if you don't explicitly set this parameter, ``setuptools``
  will include any files that match the following glob patterns:
  ``['LICEN[CS]E*', 'COPYING*', 'NOTICE*', 'AUTHORS*']``.
```

>   (setuptools 42.0.0+)

Reviewer: ```suggestion
```

See comment above. The same is also valid for the other version notes.

- All files matching the pattern ``test/test*.py``

> - ``setup.py`` (or whatever you called your setup script)

Reviewer: ```suggestion
- ``setup.py``
```

Let's remove this comment because it is not really trivial to customise the setup script name in a post-PEP517 world...

- ``setup.cfg``

> - ``README``
> - ``README.txt``
> - ``README.rst`` (Python 3.7+ or setuptools 0.6.27+)
> - ``README.md`` (setuptools 36.4.0+)

Reviewer: ```suggestion
- ``README``, ``README.txt``, ``README.rst`` or ``README.md``
```

See comment above about keeping track of versions
